main.rb

```diff
@@ -0,0 +1,191 @@
+    if !arg.nil? && arg.is_a?(Class)
+      my_each { |i| return result unless i.is_a?(arg) }
+    elsif !arg.nil? && arg.is_a?(Integer)
+      my_each { |x| return result unless x == arg }
+    elsif !arg.nil? && arg.is_a?(Regexp) || arg.is_a?(String)
+      my_each { |y| return result unless y.match(arg) }
+    elsif !block_given?
+      my_each { |a| return result if a.nil? || !a }
+    elsif block_given?
+      my_each { |b| return result unless yield(b) }
+    end
+    !result
+  end
+
+  # ========
+  #  my_any
+  # ========
+
+  def my_any?(arg = nil)
+    result = true
+
+    if !arg.nil? && arg.is_a?(Class)
+      my_each { |i| return result if i.is_a?(arg) }
+    elsif !arg.nil? && arg.is_a?(Integer)
+      my_each { |x| return result if x == arg }
+    elsif !arg.nil? && arg.is_a?(Regexp) || arg.is_a?(String)
+      my_each { |y| return result if y.match(arg) }
+    elsif !block_given?
+      my_each { |a| return result unless a.nil? || !a }
+    elsif block_given?
+      my_each { |b| return result if yield(b) }
+    end
+    !result
+  end
+
+  # ==========
+  #  my_none?
+  # ==========
+
+  def my_none?(arg = nil)
+    result = false
+
+    if !arg.nil? && arg.is_a?(Class)
+      my_each { |i| return result if i.is_a?(arg) }
+    elsif !arg.nil? && arg.is_a?(Integer)
+      my_each { |x| return result if x == arg }
+    elsif !arg.nil? && arg.is_a?(Regexp) || arg.is_a?(String)
+      my_each { |y| return result if y.match(arg) }
+    elsif !block_given?
+      my_each { |a| return result unless a.nil? || !a }
+    elsif block_given?
+      my_each { |b| return result if yield(b) }
+    end
+    !result
+  end
+
+  # ==========
+  #  my_count
+  # ==========
+
+  def my_count(count_one = nil)
+    arr_size = Array(self).length
+    count = 0
+
+    unless block_given?
+      return arr_size if count_one.nil?
+
+      arr_size.times do |i|
+        count += 1 if Array(self)[i] == count_one
+      end
+    end
+
+    if block_given? && !Array(self).nil?
+      arr_size.times do |i|
+        count += 1 if yield(Array(self)[i])
+      end
+    end
+    count
+  end
+
+  # ========
+  #  my_map
+  # ========
+
+  def my_map(proc = nil)
+    return to_enum unless block_given? || proc
+
+    new_arr = []
+    proc ? my_each { |i| new_arr << proc.call(i) } : my_each { |i| new_arr << yield(i) }
+    new_arr
+  end
+
+  # ===========
+  #  my_inject
+  # ===========
+
+  def my_inject(*arg)
+    raise 'LocalJumpError: No block or argument has been given!' unless block_given? || arg.nil?
+
+    arr = Array(self)
+
+    arg_one = arg[0]
+    arg_two = arg[1]
+    arg_two = arg_one if arg_two.nil?
+
+    if arg_one.nil? || arg_one.is_a?(Symbol)
+      arr = drop(1)
+      arg_one = to_a[0]
+    else
+      arr = to_a
+    end
+
+    if block_given?
+      arr.my_each do |i|
+        arg_one = yield(arg_one, i)
+      end
+    else
+      arr.my_each do |i|
+        arg_one = arg_one.send(arg_two, i)
+      end
+    end
+    arg_one
+  end
+end
```

Comment:
**_- `#my_inject` fails the following test cases_**
**_`array = [1, 2, 5, 2, 3, 2, 1, 3, 8, 6, 4, 4, 2, 8, 1, 1, 1, 0, 2, 4, 3, 8, 5, 7, 5, 5, 6, 6, 3, 0, 3, 6, 4, 7, 6, 4, 0, 4, 1, 0, 0, 0, 7, 7, 3, 8, 5, 5, 4, 5, 4, 1, 0, 3, 3, 0, 7, 8, 1, 2, 2, 1, 1, 8, 6, 6, 1, 7, 6, 5, 1, 0, 4, 1, 1, 2, 0, 6, 2, 1, 2, 2, 3, 2, 2, 5, 1, 4, 2, 5, 3, 4, 3, 7, 4, 0, 3, 7, 8, 5]`_**

**_`range = Range.new(5, 50)`_**

#### Using rspec for the test
**_1. `expect { array.my_inject }.to raise_error(LocalJumpError)`_**

**_2. `expect(array.my_inject(:+)).to eq array.inject(:+)`_**

**_3. `expect(range.my_inject(:*)).to eq range.inject(:*)`_**

**_4. `expect(array.my_inject(20, :*)).to eq array.inject(20, :*)`_**

**_5. `expect(range.my_inject(2, :*)).to eq range.inject(2, :*)`_**

**_6. `actual = multiply_els [2, 4, 5]      expect(actual).to eq 40`_**

Reply:
fixed and tested all your test cases, they now all pass.